cli/src/merge_tools/external.rs

```diff
@@ -478,7 +482,20 @@ pub fn invoke_external_diff(
     } else {
         cmd.current_dir(diff_dir);
     }
-    cmd.args(interpolate_variables(&tool.diff_args, &patterns));
+    let (before_args, after_args) = match &tool.diff_args.iter().position(|arg| arg == "$args") {
+        Some(args_pos) => &(
+            &tool.diff_args[0..*args_pos],
+            &tool.diff_args[*args_pos + 1..],
+        ),
+        None => &(tool.diff_args.as_slice(), &[] as &[String]),
+    };
+    cmd.args(interpolate_variables(before_args, &patterns));
+    if ui.color() {
+        cmd.args(&tool.enable_color_args);
+    } else {
+        cmd.args(&tool.disable_color_args);
+    }
+    cmd.args(interpolate_variables(after_args, &patterns));
```

Comment:
@yuja Something like this? I need to add documentation (and tests), but it injects the colour argument at an `$args` marker, or at the end of the command if there's no marker.

This assumes that all extra arguments can be injected in one spot, but that seems like a reasonable assumption.

Reply:
I would say no. It works, but looks like a hack. I would simply add separate tools, and make `diff --tool=<name> --context=N`, etc. fail or show warning.

Alternatively, maybe we can pass parameters by environment variables (e.g. `JJ_DIFF_IGNORE_ALL_SPACE=1`) which can be interpreted by shell script wrapper, but I don't think it's good idea either.

Reply:
Ah well, if this feature is too niche I'll maintain it in a private fork. Thanks for the review!